pyproject.toml

```diff
@@ -17,18 +17,33 @@
 [build-system]
 build-backend = 'mesonpy'
 requires = [
-    "meson-python>=0.15.0",
-    "Cython>=3.0.8",        # when updating version, also update check in meson.build
-    "pybind11>=2.13.2",     # when updating version, also update check in scipy/meson.build
-    "pythran>=0.14.0",
+    # The upper bound on meson-python is pre-emptive only (looser
+    # on purpose, since chance of breakage in 0.18/0.19 is low with
+    # 0.17.1 working at time of writing)
+    "meson-python>=0.15.0,<0.20.0",
+    # we need at least Cython 3.1.0a1 for free-threaded CPython
+    # 3.13 wheel builds
+    "Cython>=3.1.0a1,<3.2.0; python_version=='3.13t'",        # when updating version, also update check in meson.build
+    # for other CPython versions, the regular pre-emptive
+    # Cython version bounds policy applies
+    "Cython>=3.0.8,<3.1.0; python_version!='3.13t'",
+    # The upper bound on pybind11 is pre-emptive only
+    "pybind11>=2.13.2,<2.14.0",     # when updating version, also update check in scipy/meson.build
+    # The upper bound on pythran is pre-emptive only; 0.17.0
+    # is released/working at time of writing, and 0.18.x may
+    # have desirable free-threaded CPython 3.13 support
+    "pythran>=0.14.0,<0.19.0",
```

Comment:
given that the rest of the `pyproject.toml` is likely going to have to "ignore" free-threaded CPython, not sure the extension here makes much sense..

Reply:
Agreed. I'm fine with <0.18.0 here